### ISSUE TITLE
Possible bug?

### ISSUE THREAD

**Issue by busslina** [embraceitmobile/animated_tree_view#27]
[tree-error.webm](https://github.com/embraceitmobile/animated_tree_view/assets/73592852/e2a44ae5-6b95-4f5d-b30f-ffb9a01c1687)

**Comment by jawwad-hassan89** [comment]
Is this related to this [PR-28](https://github.com/embraceitmobile/animated_tree_view/pull/28)?
I will look into this.

**Comment by busslina** [comment]
> Is this related to this [PR-28](https://github.com/embraceitmobile/animated_tree_view/pull/28)? I will look into this.

Not really. The thing here is that when I switch tabs, state get disposed. When re-enter on the tree tab I tried to reuse the Tree object. The root node is okay (gets reset to collapsed, but the remaining nodes that were previously get expanded, now are collapsed but its state say are expanded (you can see the `ChevronIndicator`), so I must click twice on it to get well expanded.

I bypass this bug(?) by creating my own paralel Tree which save the expanded state and the children values. So when I re-enter on the Tree tab, I create a new Tree (your Tree) and re-install the state saved on my tree.

**Comment by StoneColdRepins** [comment]
@busslina can you elaborate on your solution to bypass the bug(?) or post a code sippet

**Comment by busslina** [comment]
I just simply have a parallel tree in which I save the state. When I revisit this tab, I recreate the node states based on this parallel tree.

Seems that your node implementation is not prepared for a scenario where the page/container gets disposed and rebuilt again. They save the expanded state. But when they get recreated, they are not expanded but still have the expanded state.

I could pass my code, but it is just a patch. I recommend you to review your code and when the tree is re/build check the nodes and expand them if they are marked as exoanded

**Comment by StoneColdRepins** [comment]
hi please send the patch, i am new to flutter and using the control in my app

**Comment by busslina** [comment]
> hi please send the patch, i am new to flutter and using the control in my app

https://gist.github.com/busslina/e60a41ceca418151c491a730f12a57b0

If you want help to understand my code, ask me in the Gist

**Comment by djuelg** [comment]
Hi @jawwad-hassan89, have you looked into this? It still seems to be an issue in the current version.
